### PR TITLE
 qemu: support vfio pass x-pci-vendor-id and x-pci-device-id pass 

### PR DESCRIPTION
**Pull request by Ace-Tang**
since some vendor id like 1ded can not be identified by virtio-pci
driver, so need to pass a specified vendor id to qemu.

Fixes: #1894

### CODE REVIEW

**Comment by devimc** [comment]
/test

thanks @Ace-Tang

File: virtcontainers/physical_endpoint.go [hunk 1]
```diff
@@ -79,8 +79,16 @@ func (endpoint *PhysicalEndpoint) Attach(h hypervisor) error {
 	}
 
 	// TODO: use device manager as general device management entrance
+	var vendorID, deviceID string
+	if splits := strings.Split(endpoint.VendorDeviceID, " "); len(splits) == 2 {
```
Comment (jodh-intel) [inline virtcontainers/physical_endpoint.go]: What if this test fails? `vendorID` and `deviceID` are added with blank values?

Comment (Ace-Tang) [reply virtcontainers/physical_endpoint.go]: Oh, I should add this empty check in `appendVFIODevice`

Comment (Ace-Tang) [reply virtcontainers/physical_endpoint.go]: I checked that I have add this empty check in govmm package
```
if isVirtioPCI[driver] {
		if vfioDev.VendorID != "" {
			deviceParams = append(deviceParams, fmt.Sprintf(",x-pci-vendor-id=%s", vfioDev.VendorID))
		}
		if vfioDev.DeviceID != "" {
			deviceParams = append(deviceParams, fmt.Sprintf(",x-pci-device-id=%s", vfioDev.DeviceID))
		}
```
so it is ok to add directly, the test also can pass(current fail is my signed-off-by not right) and I have add a new test to this.

Do you ok with this :smile:  @jodh-intel

**Comment by Ace-Tang** [comment]
Could this pr be promoted, or the vender need to do some update, seems some vendor is useless

**Comment by devimc** [comment]
restarting vsocks and rhel CIs

**Pull request merged**